src/ucp/core/ucp_worker.c

```diff
@@ -1713,12 +1691,66 @@ static ucs_status_t ucp_worker_init_mpools(ucp_worker_h worker)
         goto err_reg_mp_cleanup;
     }
 
+    for (iface_id = 0; iface_id < worker->num_ifaces; ++iface_id) {
+        if_attr           = &worker->ifaces[iface_id]->attr;
+        max_mp_entry_size = ucs_max(max_mp_entry_size,
+                                    if_attr->cap.am.max_short);
+        max_mp_entry_size = ucs_max(max_mp_entry_size,
+                                    if_attr->cap.am.max_bcopy);
+        max_mp_entry_size = ucs_max(max_mp_entry_size,
+                                    if_attr->cap.am.max_zcopy);
+    }
+
+    /* Create memory pools for incoming UCT messages without a UCT descriptor.
+     * Ignore sizes which are larger than maximal transport segment. */
+    worker->am_mps_map = (max_mp_entry_size == 0) ? 0 :
+                         (ucs_roundup_pow2(max_mp_entry_size) - 1) &
```

Comment:
why need -1 ?

Reply:
we have to cut `context->config.mpools_map` by `max_mp_entry_size`